vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceOverviewManager.java

```diff
@@ -0,0 +1,298 @@
+     * To map each space to all linked space
+     */
+    @Override
+    public Map<String, List<SpaceLinkDisplay>> getSpaceToSpaceLinks() {
+
+        Map<String, List<SpaceLinkDisplay>> spaceToSpaceLinksMap = new HashMap<>();
+        Iterable<SpaceLinkDisplay> spaceToSpaceLinksList = spaceLinkDisplayRepository.findAll();
+        for (SpaceLinkDisplay link : spaceToSpaceLinksList) {
+            String spaceId = null;
+            if (link.getLink() != null && link.getLink().getSourceSpace() != null
+                    && link.getLink().getSourceSpace().getId() != null) {
+
+                spaceId = link.getLink().getSourceSpace().getId();
+            }
+            if (spaceId != null) {
+                if (!spaceToSpaceLinksMap.containsKey(spaceId)) {
+                    spaceToSpaceLinksMap.put(spaceId, new ArrayList<>());
+                }
+                spaceToSpaceLinksMap.get(spaceId).add(link);
+            }
+        }
+        return spaceToSpaceLinksMap;
+    }
+
+    /**
+     * To retrieve all space to space and space to module links corresponding to
+     * each space
+     * 
+     * @param contextPath           The contextpath of the application is being
+     *                              passed
+     * @param mapper                Passing this object to create object node for
+     *                              creating json of space and module node in the
+     *                              spaceoverview graph
+     * @param nodeArray             To create the json for each node of
+     *                              spaceoverview graph
+     * @param spaceLinkMap          For holding list of all links corresponding to
+     *                              each space
+     * @param spaceToModuleLinksMap For getting list of links from one space to
+     *                              other modules
+     * @param spaceToSpaceLinksMap  For getting list of links from one space to
+     *                              other spaces
+     */
+    @Override
+    public void getSpaceLinkMap(String contextPath, ObjectMapper mapper, ArrayNode nodeArray,
+            Map<String, List<String>> spaceLinkMap, Map<String, List<ModuleLinkDisplay>> spaceToModuleLinksMap,
+            Map<String, List<SpaceLinkDisplay>> spaceToSpaceLinksMap) {
+
+        Iterable<Space> allSpacesList = spaceRepo.findAll();
+        if (allSpacesList != null) {
+            createSpaceNodeInOverviewGraph(contextPath, mapper, nodeArray);
+            for (Space space : allSpacesList) {
+                List<ModuleLinkDisplay> spaceToModulelinksList = spaceToModuleLinksMap.get(space.getId());
+                List<SpaceLinkDisplay> spaceLinks = spaceToSpaceLinksMap.get(space.getId());
+                List<SpaceLinkDisplay> spaceToSpaceLinksList = new ArrayList<>();
+                if (spaceLinks != null) {
+                    spaceToSpaceLinksList.addAll(spaceLinks);
+                }
+                List<String> spaceAllLink = new ArrayList<>();
```

Comment:
this list is called `spaceAllLink` but is a list of strings. what does it actually hold? name should reflect that.